packages/editor/src/browser/editor-navigation-contribution.ts

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable, postConstruct } from 'inversify';
+import { ILogger } from '@theia/core/lib/common/logger';
+import { StorageService } from '@theia/core/lib/browser/storage-service';
+import { Disposable, DisposableCollection } from '@theia/core/lib/common/disposable';
+import { FrontendApplicationContribution } from '@theia/core/lib/browser/frontend-application';
+import { Command, CommandContribution, CommandRegistry } from '@theia/core/lib/common/command';
+import { KeybindingContribution, KeybindingRegistry } from '@theia/core/lib/browser/keybinding';
+import { MenuContribution, MenuModelRegistry, MAIN_MENU_BAR } from '@theia/core/lib/common/menu';
+import { EditorWidget } from './editor-widget';
+import { EditorManager } from './editor-manager';
+import { TextEditor, Position, Range, TextDocumentChangeEvent } from './editor';
+import { NavigationLocation } from './navigation/navigation-location';
+import { NavigationLocationService } from './navigation/navigation-location-service';
+
+@injectable()
+export class EditorNavigationContribution implements Disposable, MenuContribution, CommandContribution, KeybindingContribution, FrontendApplicationContribution {
```

Comment:
We have `EditorContribution`. Would it make sense to merge this contribution into it? I think it is nice to have all constants for commands, menus provided by the extensions in one place.

I don't mean to merge completely `EditorContribution` only registration of commands, menus and so on. Implementation can still live in something like `EditorNavigationService` and be called from `EditorContribution`.

What do you think?

Reply:
I am OK with both ways. If you prefer moving the editor navigation contribution there, we can do it.

Reply:
👍 Sure, as I said, I am OK with both, if you feel it is better, we can do it. It's a minor work.